android/src/main/java/com/pw/droplet/braintree/Braintree.java

```diff
@@ -1,301 +1,316 @@
-                    if (error instanceof ErrorWithResponse) {
-                        ErrorWithResponse errorWithResponse = (ErrorWithResponse) error;
-                        BraintreeError cardErrors =
-                                errorWithResponse.errorFor("creditCard");
-                        if (cardErrors != null) {
-                            Gson gson = new Gson();
-                            final Map<String, String> errors = new HashMap<>();
-                            BraintreeError numberError = cardErrors.errorFor("number");
-                            BraintreeError cvvError = cardErrors.errorFor("cvv");
-                            BraintreeError expirationDateError =
-                                    cardErrors.errorFor("expirationDate");
-                            BraintreeError postalCode = cardErrors.errorFor("postalCode");
-
-                            if (numberError != null) {
-                                errors.put("card_number", numberError.getMessage());
-                            }
-
-                            if (cvvError != null) {
-                                errors.put("cvv", cvvError.getMessage());
-                            }
-
-                            if (expirationDateError != null) {
-                                errors.put("expiration_date", expirationDateError.getMessage());
-                            }
-
-                            // TODO add more fields
-                            if (postalCode != null) {
-                                errors.put("postal_code", postalCode.getMessage());
-                            }
-
-                            nonceErrorCallback(gson.toJson(errors));
-                        } else {
-                            nonceErrorCallback(errorWithResponse.getErrorResponse());
+                        break;
+                    case BraintreeRequestCodes.VENMO:
+                        if (mVenmoClient != null) {
+                            mVenmoClient.onBrowserSwitchResult(
+                                    browserSwitchResult,
+                                    this::handleVenmoResult
+                            );
                         }
-                    }
+                        break;
                 }
-            });
-            this.setToken(token);
-            successCallback.invoke(this.getToken());
+            }
         }
     }
 
-    public void nonceCallback(String nonce) {
-        if (nonce != null) {
-            this.successCallback.invoke(nonce);
-        } else {
-            this.errorCallback.invoke("Sorry, we cannot access you payment method.");
+    @ReactMethod
+    public void setup(final String token) {
+        if (mBraintreeClient == null || !token.equals(mToken)) {
+            mCurrentActivity = (FragmentActivity) getCurrentActivity();
+            mBraintreeClient = new BraintreeClient(mContext, token);
+
+            new DataCollector(mBraintreeClient).collectDeviceData(
```

Comment:
Current params is depreciated need to pass in DataCollectorRequest

```
Fixes Google Play Store Rejection
Add DataCollectorRequest to pass in hasUserLocationConsent
Update DataCollector.collectDeviceData() to take in DataCollectorRequest
Deprecate existing DataCollector.collectDeviceData()
```

![image](https://github.com/StronkMan/react-native-braintree-no-ui/assets/165301863/326f609d-eef7-4245-b502-368a0caf336d)